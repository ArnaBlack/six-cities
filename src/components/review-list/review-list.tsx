import * as React from 'react';
import {connect} from 'react-redux';

import Review from '../review/review';

import {loadReviews} from '../../store/reviews/operation/operation';
import {getReviews} from '../../store/reviews/selectors';

import {Comment} from '../../types';

interface Props {
  reviews: Comment[],
}

const ReviewList = (props: Props) => {
  const {reviews} = props;

  return <ul className="reviews__list">
    {reviews.map((it) => <Review
      key={`review-${it.id}`}
      {...it}
    />)}
  </ul>;
};

const mapStateToProps = (state, props) => ({
  ...props,
  reviews: getReviews(state),
});

export {ReviewList};
export default connect(mapStateToProps)(ReviewList);
