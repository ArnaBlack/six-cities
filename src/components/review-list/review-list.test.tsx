import * as React from 'react';
import {
  configure,
  shallow,
} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {ReviewList} from './review-list';

configure({adapter: new Adapter()});

const mock = {
  reviews: [
    {
      id: 1,
      user: {
        avatarUrl: ``,
        id: 1,
        isPro: false,
        name: ``,
      },
      rating: 4,
      comment: ``,
      date: ``,
    },
  ],
};

it(`ReviewList correctly renders`, () => {
  const {reviews} = mock;
  const tree = shallow(<ReviewList reviews={reviews} />);

  expect(tree).toMatchSnapshot();
});
