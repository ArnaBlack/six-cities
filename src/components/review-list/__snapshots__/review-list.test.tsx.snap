// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList correctly renders 1`] = `
<ul
  className="reviews__list"
>
  <Review
    comment=""
    date=""
    id={1}
    key="review-1"
    rating={4}
    user={
      Object {
        "avatarUrl": "",
        "id": 1,
        "isPro": false,
        "name": "",
      }
    }
  />
</ul>
`;
